tests/manage/z_cluster/test_delete_rook_ceph_mon_pod.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import pytest
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework.testlib import E2ETest, tier2
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@pytest.mark.polarion_id("OCS-2481")
+class TestDeleteRookCephMonPodE2EWorkloads(E2ETest):
+    """
+    Tries to delete rook-ceph-operator pod.
+    This operation creates a new pod 'rook-ceph-detect-version'
+    Try to delete the 'rook-ceph-detect-version' pod while is created
+
+    Note, this test performs the operations 5 times to get better odds
+    since it's a race issue
+    """
+
+    def test_delete_rook_ceph_mon_pod_e2e_wl(self):
+        for i in range(5):
+            rook_operator_pods = pod.get_pods_having_label(
+                label=constants.OPERATOR_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            assert len(rook_operator_pods) > 0, "No rook operator pod found"
+            rook_operator_pod_name = rook_operator_pods[0].get("metadata").get("name")
+            rook_operator_obj = pod.get_pod_obj(
+                rook_operator_pod_name, constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+
+            log.info(f"Found rook-operator pod {rook_operator_pod_name}. Deleting it.")
+
+            operator_deleted = rook_operator_obj.delete(wait=False)
+            assert operator_deleted, f"Failed to delete pod {rook_operator_pod_name}"
+            # from pdb import set_trace
+            # set_trace()
+
+            try:
+                for pod_list in TimeoutSampler(30, 1, self.get_detect_version_pod):
+                    if len(pod_list) > 0:
+                        self.rook_detect_pod_name = (
+                            pod_list[0].get("metadata").get("name")
+                        )
+                        self.rook_detect_pod_obj = pod.get_pod_obj(
+                            self.rook_detect_pod_name,
+                            constants.OPENSHIFT_STORAGE_NAMESPACE,
```

Comment:
Do you think it will be good to return pod obj in get_detect_version_pod. Doing that way we can simply call pod_obj.name to get the pod name